heartbeat/monitors/active/tcp/tcp_test.go

```diff
@@ -202,3 +202,32 @@ func TestUnreachableEndpointJob(t *testing.T) {
 		event.Fields,
 	)
 }
+
+// TestNXDomain tests that non-existent domains are correctly pinged.
+// Note, this depends on the system having a DNS resolver that doesn't hijack NXDOMAIN Results.
+func TestNXDomain(t *testing.T) {
+	host := "notadomainatallforsure.notadomain.notatldreally"
+	port := uint16(1234)
+	event := testTCPCheck(t, host, port)
+
+	dialErr := fmt.Sprintf("lookup %s", host)
+	mapvaltest.Test(
+		t,
+		mapval.Strict(mapval.Compose(
```

Comment:
I verified that the generated fields are correct against heartbeat 6.4 before the bug was introduced.

Heartbeat 7.0 has a more sane field generation approach, not dropping fields like `monitor.port` just because an error occurred during the DNS lookup phase.

Reply:
So this bug is in 6.5 and 6.6?

Reply:
Nope! I validated that it's not in 6.6 either. I just happened to have 6.4 handy to test the structure. I'm sure 6.6 is fine.